packages/fury-adapter-remote/test/fury-test.js

```diff
@@ -0,0 +1,216 @@
+        expect(result).to.be.instanceof(fury.minim.elements.ParseResult);
+        expect(result.api.title.toValue()).to.equal('swg');
+      });
+    });
+
+    it('valid swagger with autodetect', () => {
+      fury.parse({ source: swaggerSource }, (err, result) => {
+        expect(result).to.be.instanceof(fury.minim.elements.ParseResult);
+        expect(result.api.title.toValue()).to.equal('swg');
+      });
+    });
+
+    it('invalid swagger with mediaType', () => {
+      fury.parse({ source: invalidSwaggerSource, mediaType: 'application/swagger+json' }, (err, result) => {
+        expect(result).to.be.instanceof(fury.minim.elements.ParseResult);
+        expect(result.length).to.be.equal(1);
+        expect(result.annotations.length).to.be.equal(1);
+        expect(result.annotations.get(0).content).to.be.equal('API version number must be a string (e.g. "1.0.0") not a number.');
+      });
+    });
+  });
+
+  describe('#validate', () => {
+    it('blueprint with mediaType', () => {
+      fury.validate({ source: blueprintSource, mediaType: 'text/vnd.apiblueprint' }, (err, result) => {
+        expect(result).to.be.instanceof(fury.minim.elements.ParseResult);
+        expect(result.length).to.be.equal(0);
+        expect(result.annotations.length).to.be.equal(0);
+      });
+    });
+
+    it('blueprint with autodetect', () => {
+      fury.validate({ source: blueprintSource }, (err, result) => {
+        expect(result).to.be.instanceof(fury.minim.elements.ParseResult);
+        expect(result.length).to.be.equal(0);
+        expect(result.annotations.length).to.be.equal(0);
+      });
+    });
+
+    it('valid swagger with mediaType', () => {
+      fury.validate({ source: swaggerSource, mediaType: 'application/swagger+json' }, (err, result) => {
+        expect(result).to.be.instanceof(fury.minim.elements.ParseResult);
+        expect(result.length).to.be.equal(0);
+        expect(result.annotations.length).to.be.equal(0);
+      });
+    });
+
+    it('valid swagger with autodetect', () => {
+      fury.validate({ source: swaggerSource }, (err, result) => {
+        expect(result).to.be.instanceof(fury.minim.elements.ParseResult);
+        expect(result.length).to.be.equal(0);
+        expect(result.annotations.length).to.be.equal(0);
+      });
+    });
+
+    it('invalid swagger with mediaType', () => {
+      fury.parse({ source: invalidSwaggerSource, mediaType: 'application/swagger+json' }, (err, result) => {
+        expect(result).to.be.instanceof(fury.minim.elements.ParseResult);
+        expect(result.length).to.be.equal(1);
+        expect(result.annotations.length).to.be.equal(1);
+        expect(result.annotations.get(0).content).to.be.equal('API version number must be a string (e.g. "1.0.0") not a number.');
+      });
+    });
+  });
+
+  describe('#serialize', () => {
+    it('ParseResult to blueprint', () => {
+      fury.serialize({ api: blueprintApi }, (err, result) => {
```

Comment:
I don't think this test is correct given the API for Fury. `fury.serialize` accepts an "API" element,  (i,e instance of `Category`) and not a JSON Parse Result.

Take a look at the tests inside the API Blueprint Serializer at https://github.com/apiaryio/api-elements.js/blob/1cfc40a32e682d0fb236e5cd9ff12dded4088a44/packages/fury-adapter-apib-serializer/test/adapter-test.js#L32-L38 which demonstrate how the API is used.

Something like the following would work in the test where we are deserialising the JSON structure into fury elements:

```js
const api = fury.minim.serialiser.deserialise({
  element: 'category',
  meta: {
    classes: {
      element: 'array',
      content: [
        {
          element: 'string',
          content: 'api',
        },
      ],
    },
    title: {
      element: 'string',
      content: 'API',
    },
  },
  content: [],
});
```

This will uncover the implementation doesn't quite work for the expected input, you will need to update the logic to serialise the Category element:

```
  serialize({ api, minim }, cb) {
    const contentType = 'application/vnd.refract.parse-result+json; version=1.0';
    const content = minim.serialiser.serialise(api);

    axios({
      method: 'post',
      url: this.options.serializeEndpoint,
      baseURL: this.options.url,
      data: content,
      headers: {
        'Content-Type': contentType,
      },
```

Reply:
I think the interface can be unclear so I've documented it at https://github.com/apiaryio/api-elements.js/pull/183

Reply:
I use plutonium documentation https://plutonium.docs.apiary.io/#reference/0/api-description-parser/compose where is written 

> API Elements is supported, you can send an API Elements API Category to the composer using the `application/vnd.refract+json` content type. You may also send an API Elements Parse Result as `application/vnd.refract.parse-result+json`.

So I understand both MediaTypes are valid as input and both are equivalent.